activation/handler_v2.go

```diff
@@ -594,7 +637,12 @@ func (h *HandlerV2) syntacticallyValidateDeps(
 		if err != nil {
 			return nil, nil, fmt.Errorf("invalid poet membership: %w", err)
 		}
-		minLeaves = min(leaves, minLeaves)
+		nipostSizes[i].ticks = leaves / h.tickSize
+	}
+
+	totalEffectiveNumUnits, totalWeight, err := nipostSizes.sumUp()
```

Comment:
Yes, I split it on purpose  to do the light checks first (Post verification is heavy and its done last), it also simplifies tests (it's easier to do mocked expectations when checks are done "by categories").

It could be combined, although I'm not certain if there would be any performance difference and in which direction. For example, it _could_ (totally guessing without benchmarks :shrug:) turn out that calling `validator.Post()` in a tight loop uses the CPU cache better than if there were other checks in between (the single loop case).